### DISCUSSION TITLE
Breakable constraints

### DISCUSSION THREAD

**Discussion by jankrassnigg** [jrouwe/JoltPhysics#141] [Q&A] [answered]
On Constraint::SetEnabled() the documentation says:
"This can e.g. be used to implement a breakable constraint by detecting that the constraint impulse went over a certain limit and then disabling the constraint"

How do I query the impulse on a constraint? So far I haven't found anything where such information is accessible.

**Comment by jrouwe** [comment, marked as answer]
You would use e.g. HingeConstraint::GetTotalLambdaPosition/GetTotalLambdaRotation. You can see this value as the linear/angular impulse applied at the constraint in the last physics step to keep the constraint together. At the end of each physics step you would look over all your breakable constraints and see if this value exceeds a predefined threshold. If it does you would do Constraint::SetEnabled(false) or remove the constraint.
